Migrate postPageMessage functions with boolean param from testRunner.idl to testRunnerJS

colelao opened the pull request:
#### b5b21b534443eca166f7fecaa7e62619a05efcb2
<pre>
Migrate postPageMessage functions with boolean param from testRunner.idl to testRunnerJS
<a href="https://bugs.webkit.org/show_bug.cgi?id=298279">https://bugs.webkit.org/show_bug.cgi?id=298279</a>
<a href="https://rdar.apple.com/159710985">rdar://159710985</a>

Reviewed by Alex Christensen.

This migrates a large batch of functions that use postPageMessage with a boolean parameter from testRunner.idl to testRunnerJS in the TestController.

* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setRejectsProtectionSpaceAndContinueForAuthenticationChallenges): Deleted.
(WTR::TestRunner::setHandlesAuthenticationChallenges): Deleted.
(WTR::TestRunner::setShouldLogCanAuthenticateAgainstProtectionSpace): Deleted.
(WTR::TestRunner::setShouldLogDownloadCallbacks): Deleted.
(WTR::TestRunner::setShouldDownloadContentDispositionAttachments): Deleted.
(WTR::TestRunner::setShouldLogDownloadSize): Deleted.
(WTR::TestRunner::setShouldLogDownloadExpectedSize): Deleted.
(WTR::TestRunner::setBlockAllPlugins): Deleted.
(WTR::TestRunner::setShouldDecideNavigationPolicyAfterDelay): Deleted.
(WTR::TestRunner::setShouldDecideResponsePolicyAfterDelay): Deleted.
(WTR::TestRunner::setNavigationGesturesEnabled): Deleted.
(WTR::TestRunner::setIgnoresViewportScaleLimits): Deleted.
(WTR::TestRunner::setUseDarkAppearanceForTesting): Deleted.
(WTR::TestRunner::setShouldDownloadUndisplayableMIMETypes): Deleted.
(WTR::TestRunner::setShouldAllowDeviceOrientationAndMotionAccess): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
(WTR::TestRunner::shouldFinishAfterDownload const):
(WTR::TestRunner::shouldDecideNavigationPolicyAfterDelay const):
(WTR::TestRunner::shouldDecideResponsePolicyAfterDelay const):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::if):
(WTR::CompletionHandler&lt;void):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):

Canonical link: <a href="https://commits.webkit.org/299520@main">https://commits.webkit.org/299520@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/deed870e91be2eea4182c883c5958e4ec47e38a4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119271 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38955 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29609 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125507 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71340 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121148 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39652 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47537 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90616 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122223 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31639 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106929 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71035 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/30682 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25041 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69157 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101091 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25229 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128516 "Hash deed870e for PR 50212 does not build (failure)") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46185 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/34943 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/128516 "Hash deed870e for PR 50212 does not build (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46550 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103135 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/128516 "Hash deed870e for PR 50212 does not build (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44434 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22439 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/42756 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18984 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46052 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45515 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48865 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47202 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
```diff
@@ -371,14 +363,7 @@ class TestRunner : public JSWrappable {
     JSValueRef neverInlineFunction(JSContextRef, JSValueRef function);
 
     bool shouldDecideNavigationPolicyAfterDelay() const { return m_shouldDecideNavigationPolicyAfterDelay; }
```
Could you remove this and m_shouldDecideNavigationPolicyAfterDelay?  It's now unused.

achristensen07 commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -2031,6 +2046,81 @@ void TestController::didReceiveScriptMessage(WKScriptMessageRef message, Complet
+        return completionHandler(nullptr);
+    }
+
+    if (WKStringIsEqualToUTF8CString(command, "setRejectsProtectionSpaceAndContinueForAuthenticationChallenges")) {
```
Just for consistency, could you capitalize the S?

colelao replied:
Good catch

colelao committed:
Migrate postPageMessage functions with boolean param from testRunner.idl to testRunnerJS
https://bugs.webkit.org/show_bug.cgi?id=298279
rdar://159710985

Reviewed by Alex Christensen.

This migrates a large batch of functions that use postPageMessage with a boolean parameter from testRunner.idl to testRunnerJS in the TestController.

* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setRejectsProtectionSpaceAndContinueForAuthenticationChallenges): Deleted.
(WTR::TestRunner::setHandlesAuthenticationChallenges): Deleted.
(WTR::TestRunner::setShouldLogCanAuthenticateAgainstProtectionSpace): Deleted.
(WTR::TestRunner::setShouldLogDownloadCallbacks): Deleted.
(WTR::TestRunner::setShouldDownloadContentDispositionAttachments): Deleted.
(WTR::TestRunner::setShouldLogDownloadSize): Deleted.
(WTR::TestRunner::setShouldLogDownloadExpectedSize): Deleted.
(WTR::TestRunner::setBlockAllPlugins): Deleted.
(WTR::TestRunner::setShouldDecideNavigationPolicyAfterDelay): Deleted.
(WTR::TestRunner::setShouldDecideResponsePolicyAfterDelay): Deleted.
(WTR::TestRunner::setNavigationGesturesEnabled): Deleted.
(WTR::TestRunner::setIgnoresViewportScaleLimits): Deleted.
(WTR::TestRunner::setUseDarkAppearanceForTesting): Deleted.
(WTR::TestRunner::setShouldDownloadUndisplayableMIMETypes): Deleted.
(WTR::TestRunner::setShouldAllowDeviceOrientationAndMotionAccess): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
(WTR::TestRunner::shouldFinishAfterDownload const):
(WTR::TestRunner::shouldDecideNavigationPolicyAfterDelay const):
(WTR::TestRunner::shouldDecideResponsePolicyAfterDelay const):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::if):
(WTR::CompletionHandler<void):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):

Canonical link: https://commits.webkit.org/299520@main
